src/core/src/op/util/nms_base.cpp

```diff
@@ -128,7 +132,7 @@ void ov::op::util::NmsBase::validate_and_infer_types() {
 
     // 'selected_outputs' have the following format:
     //      [number of selected boxes, [class_id, box_score, xmin, ymin, xmax, ymax]]
-    set_output_type(0, element::f32, {first_dim_shape, 6});
+    set_output_type(0, get_input_element_type(0), {first_dim_shape, 6});
```

Comment:
please correct the specification. Now the logic contradicts the specification.
@jane-intel, fyi.

Reply:
Updated.